Add trading framework concepts page (#113)
vegaprotocol/documentation: docs/concepts/trading-on-vega/positions-margin.md

Margin is the amount of assets required to keep your positions open and orders funded. You can think of margin as the 'down payment' to open a position. Leverage, meanwhile, describes how many times larger in value than your margin you can have as a position. (For example, if you need 20DAI to open a position worth 100DAI: Your leverage is 5x and your initial margin is 20% of the full value.)

The exact margin requirements of open orders and positions are determined by the market's risk model and market conditions. The larger the position and the more volatile the market, the greater the amount of margin that will be set aside. The volatility tolerance of the market is driven by the risk model.

## Positions
A trader's position on a market is their open volume and orders. Margin requirements are calculated based on the position, and recalculated based on market movements and changes to the position. Orders that the increase your risk level of your position will require more margin.

> When a party on Vega opens a position, the minimum amount of money required to open that position is put into a margin account for that party in that market. Over the course of the position's lifetime, the margin requirements will likely change - the margin account may be topped up, and/or some margin is released back to collateral. If that party is trading on more than one market that uses the same asset, the collective positions on those markets will inform how much is set aside for margin. 

Reviewer: I wonder if this is a good paragraph to introduce the word "initial". as in
```suggestion
When a party on Vega opens a position, the minimum amount of money required to open that position is put into a margin account for that party in that market. This is referred to as the **"initial"** margin. Over the course of the position's lifetime, the margin requirements will likely change - the margin account may be topped up, and/or some margin is released back to collateral. If that party is trading on more than one market that uses the same asset, the collective positions on those markets will inform how much is set aside for margin. 
```

Reviewer: > If that party is trading on more than one market that uses the same asset, the collective positions on those markets will inform how much is set aside for margin. 
I think this might/could imply that that more margin is set aside if you have 2 positions with one key than if you had two positions with two keys.

## Automated market mechanisms 
As markets and collateral are not managed through human intervention, markets must have certain automated processes that allow for well-functioning markets and assurance that the collateral required to manage positions is available when it's needed.

There are a few mechanisms that work differently to how they would on a centralised exchange, in order to keep the markets solvent. They include:

> - [**Mark to market**](#mark-to-market): Mark to market on Vega happens much more frequently than on a centralised exchange. Every time the market price moves, the mark to market price is recalculated

Reviewer: ```suggestion
- [**Mark to market**](#mark-to-market): Mark to market on Vega happens much more frequently than on an exchange where counter parties are not pseudonymous. Every time the market price moves, the mark to market price could be recalculated
```
Could be because I think in the future there may be models in place where it doesn't need to be

> - [**Margin**](#margin): Vega has implemented automated [cross-margining](#cross-margining). Margin is calculated automatically depending on the number of positions, the size, and the market movements so that there's enough collateral available to sustain a position and not put other market participants, or the market itself, under strain

Reviewer: ```suggestion
- [**Margin**](#margin): When a party opens a position: the **initial** margin requirement is calculated automatically depending on the risk model of the market. If the market moves against the party, towards the **Maintenance** level, Vega will **Search** for more collateral in the general account, to avoid liquidation. Margin can also be **Released** if the position is in sufficient profit. Other positions in markets with the same settlement asset may also search the same general account. Therefor Vega is a cross-margin based system by default. Users can isolate margin by trading with different keys
```

Reviewer: ☝️ This a big re-write. feel free to take only what you want from it, if anything.

Feels weird that "margin" is just one bullet point in this file though. could get its own section.

When marking to market, the protocol takes the current market price and recalculates traders' margin requirements based on how their position is affected by price moves.

If the market price goes up, a trader that holds a long position receives money in their margin account – equal to the underlying's change in value – from a trader that holds a short position, and conversely if the value goes down, the holder of the short position recieves money from the holder of the long position.

For a futures market created on Vega, the mark-to-market price is calculated every time the price moves, and is based on the last traded price. This is in contrast to traditional futures markets, for which marking to market occurs once per day. One exception is when the market settles at expiry, at which point the mark to market price comes from the data source's final settlement price.

Settlement instructions are generated based on the change in market value of the open positions of a party. When the mark price changes, the network calculates settlement cash flows for each party, and the process is repeated each time the mark price changes until the maturity date for the market is reached.

> Because the margin for a market is calculated dynamically based on the market conditions, the mark price also has an effect on how much collateral is set aside for margin.

Reviewer: I'm not sure what this sentence is doing in addition to the rest of the section. Which I think is quite tight

:::note Read more
[Mark to market settlement](./settlement#mark-to-market-settlement)
:::

## Margin

> Margin is the amount of collateral required to keep your position open. It can change depending on how your position is impacted by your own actions and market movement. The margin calculations ensure a trader does not enter a trade that will immediately need to be closed out.

Reviewer: I think the last sentence in this para should be part of the next para.

> The calculation for a new position, and the amount deducted from collateral to cover margin, is based on all of a trader's open orders. A trader will need enough margin to keep a position open, whether it goes for or against the trader.

Reviewer: ```suggestion
When placing a key's first order on a market: the protocol will calculate the initial margin required. If this there is not a sufficient balance in the general account to fund this, the order will be rejected. If there is these funds will be moved into a margin account for that market.  
```
Removed the stuff about other orders, because when opening a position there wouldn't be any.

> The margin set aside for a given position is dictated by the specifics of the position and its exposure. For example, if a trader has a long position on a market, and then reduces the position - their margin requirement doesn't get larger.

Reviewer: ```suggestion

Orders that increase your open volume will increase the required margin. Orders that decrease it should not increase your margin requirements (unless you end up opening a position in the opposite direction).
```

:::note Go deeper
**[Automated cross margining](https://vega.xyz/papers/vega-protocol-whitepaper.pdf#page21)** - Section 6 of the protocol whitepaper.
:::

## Cross-margining
Vega's margining system provides automated cross margining. Cross margining, which means gains on one market can be released and used as margin on another, is supported between all markets that use the same settlement asset.

> To more closely control how much is risked on a position, it's possible to replicate the effects of isolated margin by using one party (public key) per market.

Reviewer: I've introduced some repletion with a previous suggestion but think this is okay.

  * m<sup>search</sup> := α<sup>search</sup> * m<sup>maintenance</sup>, where α<sup>search</sup> is a constant and α<sup>search</sup> > 1.
* **Release level**: Once a trader's margin balance exceeds the margin release level, the position is considered overcollateralised and funds are released back to collateral so that the margin balance is equal to the current initial margin level:
  * m<sup>release</sup> := α<sup>release</sup> * m<sup>maintenance</sup>, where α<sup>release</sup> is a constant and α<sup>release</sup> > α<sup>initial</sup>.
* **Maintenance margin**: This is implied by the risk model and corresponds to the minimum amount required to cover adverse market moves with a given probability level. As soon as the margin balance drops below the maintenance margin level, the position close-out process gets initiated.

:::note Go deeper
**[Margins and credit risk ↗](https://vega.xyz/papers/margins-and-credit-risk.pdf)** - Section 6 of the protocol whitepaper.
:::

### Calculating the margin on open orders

> The network calculates the largest long / short position. 

Reviewer: I think it is worth a reminder that "position" on Vega is open volume & open orders, in this case only the open orders that are in the same direction as your opening volume. Or to put another way.

This is not good enough to be a suggestion but:

> The network calculates the what side of your position is riskiest. 
> e.g. You have an open vol of 10 (long) with an open order to buy 5, you also have an order to sell 10. The long is riskiest (10+5).

There is an open sell order of size 1 on the book. The risk factor for short positions is 0.074347011. The current mark price is 0.02690. So minimum margin = 0.2690 x 0.074347011 = 0.00200 (rounded to 5 decimal places).

### Margin calculations on open positions
The following calculation takes into account 'slippage', as seen on an order book.

#### Example [WIP]
<img alt="Calculating margin on an open positions" src="/images/3-margin-open-positions.png" width="500" />

The trader has an open short position of size 1, and no open orders.

The risk factor for short position is 0.074347011. 

The current mark price is 0.02672. 

The best offer price is 0.02676 and it has enough volume so that theoretically the position could be closed-out at that price. 

maintenance margin = 0.02672 x 0.074347011 + max (0, 0.02676 - 0.02672) = 0.00203 (rounded to 5 decimal places), where the second term in the sum is the 'slippage' component. 

Other margin levels are derived from the maintenance margin using the scaling factors that form part of the market configuration. -->

### Initial margin calculation

> The initial margin is the minimum amount of collateral required to enter a new trade.

Reviewer: ```suggestion
The initial margin is the minimum amount of collateral required to create an order that will open a position.
```

For a futures market created on Vega, the mark-to-market price is calculated every time the price moves, and is based on the last traded price. This is in contrast to traditional futures markets, for which marking to market occurs once per day. One exception is when the market settles at expiry, at which point the mark to market price comes from the data source's final settlement price.

Settlement instructions are generated based on the change in market value of the open positions of a party. When the mark price changes, the network calculates settlement cash flows for each party, and the process is repeated each time the mark price changes until the maturity date for the market is reached.

Because the margin for a market is calculated dynamically based on the market movements, the mark price also has an effect on how much collateral is set aside for margin.

:::note Read more
[Mark to market settlement](./settlement#mark-to-market-settlement)
:::

## Margin

> Margin is the amount of collateral required to keep your position open. It can change depending on how your position is impacted by your own actions and market movement. The margin calculations ensure a trader does not enter a trade that will immediately need to be closed out.

Reviewer 2: ```suggestion
Margin is the amount of collateral required to keep your position open. It can change depending on how your position is impacted by your own actions and market movement.
```

Leverage is, in effect, the inverse of the initial margin that the protocol takes from your collateral when you place a trade. 

Margin is the amount of assets required to keep your positions open and orders funded. You can think of margin as the 'down payment' to open a position. Leverage, meanwhile, describes how many times larger in value than your margin you can have as a position. (For example, if you need 20DAI to open a position worth 100DAI: Your leverage is 5x and your initial margin is 20% of the full value.)

The exact margin requirements of open orders and positions are determined by the market's risk model and market conditions. The larger the position and the more volatile the market, the greater the amount of margin that will be set aside. The volatility tolerance of the market is driven by the risk model.

## Positions
A trader's position on a market is their open volume and orders. Margin requirements are calculated based on the position, and recalculated based on market movements and changes to the position. Orders that the increase your risk level of your position will require more margin.

> When a party on Vega opens a position, the minimum amount of money required to open that position is put into a margin account for that party in that market. 

Reviewer: ```suggestion
When a party on Vega opens a position, the minimum amount of assets required to open that position is put into a margin account for that party in that market. 
```

> Over the course of the position's lifetime, the margin requirements will likely change - the margin account may be topped up, and/or some margin is released back to collateral. If that party is trading on more than one market that uses the same asset, the collective positions on those markets will inform how much is set aside for margin. 

Reviewer 2: ```suggestion
Over the course of the position's lifetime, the margin requirements will likely change - the margin account may be topped up, and/or some margin is released back to collateral. If that party is trading on more than one market that uses the same asset, the collective positions on those markets will inform how much is set aside for margin. Margin balances are also affected by open profit and loss. 

`[margin account balance] = [initial margin requirement] + [unrealised profit] OR - [unrealised losses]`
```

> 

Reviewer: There is one thing missing (I think IMHO) from this really good intro.
it's that your margin balance is also effected by your open profit /loss. It can be thought of as 
```
[initial margin requirement]+[unrealised profit] OR - [unrealised losses]
```

## Automated market mechanisms 
As markets and collateral are not managed through human intervention, markets must have certain automated processes that allow for well-functioning markets and assurance that the collateral required to manage positions is available when it's needed.

There are a few mechanisms that work differently to how they would on a centralised exchange, in order to keep the markets solvent. They include:

> - [**Mark to market**](#mark-to-market): Mark to market on Vega happens much more frequently than on an exchange where counter parties are not pseudonymous. Every time the market price moves, the mark to market price could be recalculated

Reviewer 2: ```suggestion
```

> - [**Margin**](#margin): When a party opens a position: the *initial margin* requirement is calculated automatically depending on the market's risk model. If the market moves against the party, and the margn towards the *maintenance level*, Vega will *search* for more collateral in the general account, to avoid liquidating the position. Margin can also be *released* if the position is in sufficient profit. Other positions in markets with the same settlement asset may also interact with the same general account. Therefore, Vega is a cross-margin based system by default.

Reviewer: soft preference for leading with margin, then M2M
You can then say that mark to market is used to move assets into your margin account (from someone else's) if you are in profit, or out if not.

Reviewer 2: I only moved it around to match the order on the page, but I agree that margin being first makes more sense narratively. And the only reason margin is second is because it has so much more content. So I'll flip them around again and add your suggested sentence

> - [**Margin**](#margin): When a party opens a position: the *initial margin* requirement is calculated automatically depending on the market's risk model. If the market moves against the party, and the margn towards the *maintenance level*, Vega will *search* for more collateral in the general account, to avoid liquidating the position. Margin can also be *released* if the position is in sufficient profit. Other positions in markets with the same settlement asset may also interact with the same general account. Therefore, Vega is a cross-margin based system by default.

Reviewer 2: ```suggestion
- [**Margin**](#margin): When a party opens a position: the *initial margin* requirement is calculated automatically depending on the market's risk model. If the market moves against the party, and the margn towards the *maintenance level*, Vega will *search* for more collateral in the general account, to avoid liquidating the position. Margin can also be *released* if the position is in sufficient profit. Other positions in markets with the same settlement asset may also interact with the same general account. Therefore, Vega is a cross-margin based system by default.
- [**Mark to market**](#mark-to-market): Mark to market on Vega happens much more frequently than on an exchange where counter parties are not pseudonymous. Every time the market price moves, the mark to market price could be recalculated. Mark to market is used to move assets into your margin account (from someone else's) if you are in profit, or out of your margin account if not.
```

## Mark to market
Marking to market refers to settling gains and losses due to changes in the market value of the underlying product. Marking to market aims to provide a realistic appraisal of a position based on the current market conditions.

When marking to market, the protocol takes the current market price and recalculates traders' margin requirements based on how their position is affected by price moves.

If the market price goes up, a trader that holds a long position receives money in their margin account – equal to the underlying's change in value – from a trader that holds a short position, and conversely if the value goes down, the holder of the short position recieves money from the holder of the long position.

> For a futures market created on Vega, the mark-to-market price is calculated every time the price moves, and is based on the last traded price. This is in contrast to traditional futures markets, for which marking to market occurs once per day. One exception is when the market settles at expiry, at which point the mark to market price comes from the data source's final settlement price.

Reviewer: ```suggestion
For a futures market created on Vega, the mark-to-market price is calculated every time the price moves, and is based on the last traded price. This is in contrast to traditional futures markets, for which marking to market may occur once per day. One exception is when the market settles at expiry, at which point the mark to market price comes from the data source's final settlement price.
```

The amount a trader will have held aside as maintenance margin is derived from the market's risk model. Specifically, it's based on a risk measure called the expected shortfall, used to evaluate the market or credit risk of the collective positions.

If the margin balance drops below the maintenance margin level, the position closeout process gets initiated.

:::note Read more
[Closeouts](./market-protections#closeouts)
[Risk models](../vega-protocol#risk-models-and-parameters)
:::

### Margin level: Initial
The *initial margin level* is the amount that will be transferred from the trader's general account to be used as margin when an order is placed or a trade is executed. The initial margin is more than the absolute minimum needed to support a position, as it offers a cushion to keep a position open as the mark price changes.

The initial margin is scaled from the *maintenance margin* amount. It's calculated by multiplying the maintenance margin amount by the `initial_margin` scaling factor, which is set by the network parameter `market.margin.scalingFactors`.

> `initial margin level = maintenance margin x initial_margin scaling factor`

Reviewer 2: ```suggestion
`[initial margin level] = [maintenance margin] x [initial_margin scaling factor]`
```
